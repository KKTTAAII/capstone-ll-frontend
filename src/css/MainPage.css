.MainPage-container {
    margin: 20vh;
    display: flex;
    flex-direction: column;
}

.MainPage-header {
    /*credit https://css-tricks.com/snippets/css/typewriter-effect/*/
    overflow: hidden; /* Ensures the content is not revealed until the animation */
    border-right: .15em solid rgb(0, 0, 0); /* The typwriter cursor */
    white-space: nowrap; /* Keeps the content on a single line */
    margin: 0 auto; /* Gives that scrolling effect as the typing happens */
    letter-spacing: .15em; /* Adjust as needed */
    animation: 
    typing 3.5s steps(30, end),
    blink-caret .30s step-end infinite;
}

/* The typing effect */
@keyframes typing {
    from { width: 0 }
    to { width: 15% }
  }
  
  /* The typewriter cursor effect */
  @keyframes blink-caret {
    from, to { border-color: transparent }
    50% { border-color: rgb(0, 0, 0); }
  }