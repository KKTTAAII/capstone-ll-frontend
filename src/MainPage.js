import React from "react";
import "./css/MainPage.css";
import PawPrints from "./assets/paw-print-mainpage/pawPrint.jpg";

const MainPage = () => {
  return (
    <div className="MainPage-container">
      {/* {credit <a href="https://www.vecteezy.com/free-vector/paw-print">Paw Print Vectors by Vecteezy</a>} */}
      <img alt="pawprint" src={PawPrints} />
      <div className="MainPage-header">Welcome to Petly!</div>
    </div>
  );
};

export default MainPage;
