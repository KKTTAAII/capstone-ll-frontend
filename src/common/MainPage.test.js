import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import MainPage from "./MainPage";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <MainPage />
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <MainPage />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("shows main page text", () => {
  const { getByText } = render(
    <MemoryRouter>
      <MainPage />
    </MemoryRouter>
  );

  //expect the welcome text in the Document
  expect(getByText("Welcome to Petly!")).toBeInTheDocument();
});