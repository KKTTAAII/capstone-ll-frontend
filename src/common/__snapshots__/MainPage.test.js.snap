// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MainPage-container"
  >
    <img
      alt="pawprint"
      src="pawPrint.jpg"
    />
    <div
      class="MainPage-header"
    >
      Welcome to Petly!
    </div>
  </div>
</DocumentFragment>
`;
